connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -248,9 +249,13 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorProps) {
         Connector connector = getConnector(connType);
         ClassLoader savedLoader = plugins().compareAndSwapLoaders(connector);
         try {
-            ConfigDef baseConfigDef = (connector instanceof SourceConnector)
-                    ? SourceConnectorConfig.configDef()
-                    : SinkConnectorConfig.configDef();
+            ConfigDef baseConfigDef;
+            if (SourceConnector.class.isAssignableFrom(Class.forName(connType))) {
```

Comment:
I changed the logic for determining source or sink due to issues in testing. In the herder tests, mocked connectors show a class of `org.apache.kafka.connect.connector.Connector$$EnhancerByCGLIB$$d27185f2`, which is neither a SinkConnector nor a SourceConnector, so the existing `instanceof` check always returns false in the tests.

I'd be happy to revert this logic change if the `Class.forName(connType)` is undesirable, but I'd need help understanding why the mocks appear as `Connector` rather than the subtype that appears in the configs.